webapp/src/main/java/com/box/l10n/mojito/service/assetExtraction/AssetExtractionService.java

```diff
@@ -196,77 +200,136 @@ public class AssetExtractionService {
      * @param filterOptions
      * @param currentTask            The current task, injected
      * @return A {@link Future}
-     * @throws UnsupportedAssetFilterTypeException                                          If asset type not supported
-     * @throws java.lang.InterruptedException
-     * @throws com.box.l10n.mojito.service.assetExtraction.AssetExtractionConflictException
+     * @throws UnsupportedAssetFilterTypeException If asset type not supported
+     * @throws InterruptedException
+     * @throws AssetExtractionConflictException
      */
     public PollableFuture<Asset> processAsset(
             Long assetContentId,
+            Long pushRunId,
             FilterConfigIdOverride filterConfigIdOverride,
             List<String> filterOptions,
-            PollableTask currentTask) throws UnsupportedAssetFilterTypeException, InterruptedException, AssetExtractionConflictException {
+            PollableTask currentTask)
+            throws UnsupportedAssetFilterTypeException, InterruptedException, AssetExtractionConflictException {
 
         logger.info("Start processing asset content, id: {}", assetContentId);
         AssetContent assetContent = assetContentService.findOne(assetContentId);
         Asset asset = getUndeletedAsset(assetContent.getAsset());
 
-        MultiBranchState stateForNewContent = convertAssetContentToMultiBranchState(assetContent, filterConfigIdOverride, filterOptions, currentTask);
-        CreateTextUnitsResult createdTextUnitsResult = createTextUnitsForNewContent(assetContent, stateForNewContent, currentTask);
+        MultiBranchState stateForNewContent = convertAssetContentToMultiBranchState(assetContent,
+                                                                                    filterConfigIdOverride,
+                                                                                    filterOptions,
+                                                                                    currentTask);
+        CreateTextUnitsResult createdTextUnitsResult = createTextUnitsForNewContent(assetContent,
+                                                                                    stateForNewContent,
+                                                                                    currentTask);
 
         updateBranchAssetExtraction(assetContent, createdTextUnitsResult.getUpdatedState(), filterOptions, currentTask);
         updateLastSuccessfulAssetExtraction(asset, createdTextUnitsResult.getUpdatedState(), currentTask);
+        updatePushRun(asset, createdTextUnitsResult.getUpdatedState(), pushRunId, currentTask);
```

Comment:
@aurambaj  Is
`createdTextUnitsResult.getUpdatedState() `
the right TU source to use here - that won't pull in TUs from other branches, but will still cover existing + new TUs?

Reply:
yes, exact

Reply:
Thanks for confirming!